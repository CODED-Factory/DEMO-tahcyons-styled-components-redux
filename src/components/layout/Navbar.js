import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";

// Images
import logo from "../../images/logo.svg";

// Styles
import NavbarStyles from "./styles/NavbarStyles";

// Actions
import { logout } from "../../redux/actions";

export const Navbar = ({ user, logout }) => {
  return (
    <NavbarStyles className="bg-white shadow-1 fixed left-0 top-0 w-100">
      <div className="container flex justify-between items-center pv3">
        <Link to="/">
          <img src={logo} alt="logo" />
        </Link>
        <Link
          to={user ? "/" : "/login"}
          onClick={user && logout}
          className="b ph3 pv2 input-reset ba b--black grow pointer f6 dib"
        >
          {user ? "LOGOUT" : "LOGIN"}
        </Link>
      </div>
    </NavbarStyles>
  );
};

const mapStateToProps = ({ auth }) => ({
  user: auth.user
});

const mapDispatchToProps = {
  logout
};

export default connect(mapStateToProps, mapDispatchToProps)(Navbar);
