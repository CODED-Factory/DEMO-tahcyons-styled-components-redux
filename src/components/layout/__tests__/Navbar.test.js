import React from "react";
import { shallow, mount } from "enzyme";
import { Navbar } from "../Navbar";

describe("<Navbar />", () => {
  it("renders with a login button if there's no user", () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders with a logout button if there's a user", () => {
    const wrapper = shallow(<Navbar user={{ username: "mr potato" }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("logs the user out when the logout button is pressed", () => {
    const mockLogout = jest.fn();
    const wrapper = shallow(
      <Navbar user={{ username: "mr potato" }} logout={mockLogout} />
    );
    wrapper.find(".b").simulate("click");
    expect(mockLogout).toHaveBeenCalledTimes(1);
  });
});
