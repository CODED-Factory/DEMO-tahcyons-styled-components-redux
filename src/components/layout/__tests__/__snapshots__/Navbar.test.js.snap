// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders with a login button if there's no user 1`] = `
<styled.nav
  className="bg-white shadow-1 fixed left-0 top-0 w-100"
>
  <div
    className="container flex justify-between items-center pv3"
  >
    <Link
      to="/"
    >
      <img
        alt="logo"
        src="logo.svg"
      />
    </Link>
    <Link
      className="b ph3 pv2 input-reset ba b--black grow pointer f6 dib"
      to="/login"
    >
      LOGIN
    </Link>
  </div>
</styled.nav>
`;

exports[`<Navbar /> renders with a logout button if there's a user 1`] = `
<styled.nav
  className="bg-white shadow-1 fixed left-0 top-0 w-100"
>
  <div
    className="container flex justify-between items-center pv3"
  >
    <Link
      to="/"
    >
      <img
        alt="logo"
        src="logo.svg"
      />
    </Link>
    <Link
      className="b ph3 pv2 input-reset ba b--black grow pointer f6 dib"
      to="/"
    >
      LOGOUT
    </Link>
  </div>
</styled.nav>
`;
